src/vm/threads.cpp

```diff
@@ -8816,41 +8512,8 @@ static void ManagedThreadBase_DispatchInner(ManagedThreadCallState *pCallState)
     }
     CONTRACTL_END;
 
-
-    Thread *pThread = GetThread();
-
-    if (!pCallState->IsAppDomainEqual(pThread->GetDomain()))
-    {
-        // On Win7 and later, AppDomain transitions at the threadbase will *not* have EH setup at transition boundary.
-        // This implies that an unhandled exception from the base domain (i.e. AD in which the thread starts) will
-        // not return to DefDomain but will continue to go up the stack with the thread still being in base domain.
-        // We have a holder in ENTER_DOMAIN_*_NO_EH_AT_TRANSITION macro (ReturnToPreviousAppDomainHolder) that will
-        // revert AD context at threadbase if an unwind is triggered after the exception has gone unhandled.
-        //
```

Comment:
It looks a bit suspect that this disappeared. Is this really not needed for anything?

Reply:
The `if (!pCallState->IsAppDomainEqual(pThread->GetDomain())) ` is always false, since we only have one app domain, so I don't see how this code could be ever executed. The goal of the code in the `if` branch was only to use the AD callback for cases when the current AppDomain and the pCallState's AppDomain differ.
Or have I misunderstood what you were referring to?

Reply:
> `if (!pCallState->IsAppDomainEqual(pThread->GetDomain()))` is always false,

Is it false even if the thread have not entered the appdomain yet? In other words, can `pThread->GetDomain()` return "no appdomain"?

Reply:
I've looked through the code. If the `pThread->GetDomain()` returned NULL here, the `pCallState->IsAppDomainEqual` would crash here if we came from `ManagedThreadBase_FullTransitionWithAD` where the `pCallState` is constructed with `bDomainIsAsID` set to true  and so the `pCallState->IsAppDomainEqual` would call `GetId()` on the passed in `AppDomain *`.
The other possible call path stems from `ManagedThreadBase_NoADTransition`, where the `pCallState` is constructed with `bDomainIsAsID` set to false and the ManagedThreadCallState contains a pointer obtained via GetAppDomain(). The `ManagedThreadBase_NoADTransition` is called only from `ManagedThreadBase::FinalizerBase`, which in turn is called from `FinalizerThread::FinalizerThreadStart` and there is an assert `_ASSERTE(GetFinalizerThread()->GetDomain()->IsDefaultDomain());` before the call. So the `pThread->GetDomain()` should never return NULL.

Reply:
ok